deviceshifu/pkg/deviceshifu/deviceshifu.go

```diff
@@ -166,6 +166,19 @@ type DeviceCommandHandlerHTTP struct {
 	deviceShifuHTTPHandlerMetaData *DeviceShifuHTTPHandlerMetaData
 }
 
+func createUriFromRequest(address string, handlerInstruction string, r *http.Request) string {
+	queryStr := "?"
+	for queryName, queryValues := range r.URL.Query() {
+		for _, queryValue := range queryValues {
+			queryStr += queryName + "=" + queryValue + "&"
+		}
+	}
+
+	queryStr = strings.TrimSuffix(queryStr, "&")
+
+	return "http://" + address + "/" + handlerInstruction + queryStr
```

Comment:
check if `queryStr` is `?` before return so that we don't append `?` to all requests